jcstress-core/src/main/java/org/openjdk/jcstress/TimeBudget.java

```diff
@@ -131,10 +134,46 @@ public void printOn(PrintStream out) {
         } else {
             out.println("    Initial completion estimate: " + ReportUtils.msToDate(timeLeftMs(), true));
             out.println("    Initial test time: " + targetTestTimeMs() + " ms");
+            printOvertimeWarning();
         }
         out.println();
     }
 
+    private long countEta(int msPerTest) {
+        long expectedTotalTime = (long) expectedTests * msPerTest;
+        long expectedPerTest = expectedTotalTime / getConcurentTestsFactor();
+        return expectedPerTest;
+    }
+
+    private boolean printOvertimeWarning() {
+        long expectedPerTest = countEta(DEFAULT_PER_TEST_MS);
+        boolean print=false;
+        if (expectedPerTest > budget.milliseconds() * 2l) {
+            System.out.println(" + +++ FATAL - your tests will never finish as expected. They will run much longer ");
+            print=true;
+        }
+        if (expectedPerTest * 2 < budget.milliseconds() * 2l) {
+            System.out.println(" + +++ WARNING - your time budget will not be used. Tests will end much sooner.");
+            print=true;
+        }
+        if (print) {
+            System.out.println(" | For " + expectedTests + " with concurrency factor of " + getConcurentTestsFactor()
```

Comment:
If it was me, I would switch these `println` calls for `printf` to make the messages more readable.

Reply:
I had rewrote it to printf, but at the end found the println slightly more readable. It seems println is a bit more used then printf in the prject, so unless yuo insists I will keep println.

Reply:
No, that's fine to leave as is